src/App.tsx

```diff
@@ -1,42 +1,37 @@
-import React, { ChangeEvent, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Container } from 'react-bootstrap';
 import { TodoItem } from './types';
 import { AddItemForm, TodoList } from './components';
 
 function App() {
-  const [item, setItem] = useState('');
   const [list, setList] = useState<TodoItem[]>([]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    if (e.target.name === 'input') {
-      setItem(e.target.value);
-    }
-  };
-
-  const addTodo = () => {
+  const addTodo = (item: string) => {
     const newItem = { text: item, completed: false };
-    if (newItem.text !== '') {
+    if (newItem.text.trim() !== '') {
       setList([...list, newItem]);
-      setItem('');
+      localStorage.setItem('todoList', JSON.stringify([...list, newItem]));
```

Comment:
I think you can consider "saving" as side effect... So let's use "useEffect" hook for that:
```ts
useEffect(() => {
    // you should prevent first saving here...
    localStorage.setItem('todoList', JSON.stringify(list));
}, [list])
```